content/docs/rfcs/14/README.md

```diff
@@ -50,6 +50,8 @@ The data payload is also treated as a Waku message attribute for convenience.
 
 * The `content_topic` attribute MUST specify a string identifier that can be used for content-based filtering.
 
+* The `meta` attribute, if present, contains an arbitrary application-specific variable-length byte array with a maximum length limit of 32 bytes.
```

Comment:
> And explanation why 32 bytes have been chosen as the length would be helpful.
> Why is not larger to potentially allow holding a signature (+ULID)?

Hmmm, that's tricky. I would prefer not to give any ideas about the field's content here.   My idea is to keep this section as simple as possible.

> Imo, it would make sense if this single `meta` attribute holds all info an application might want to hand down to the relay layer.

There is a size trade-off.

The 32 bytes size is big enough to fit a ULID (16 bytes) plus 16 bytes of application info (e.g, `[4x u32]`, `[u64, 8x char]`, etc.). And it is small enough to only increase the gossip control messages sizes by 32 bytes (at this moment, we are already gossiping a 32 bytes sha256 of the message protobuf).

A bigger size would be problematic as this field will also be gossiped via the `IWANT/IHAVE` control messages. It will form part of the _Message Unique ID_ (64 bytes = 32 bytes sha256 + 32 bytes meta).

For example, allowing a 128 bytes (1kiB) `meta` field will increase the bandwidth requirements for this "control field". This impacts the bandwidth requirements of Gossipsub in bandwidth-restricted environments. And additionally, there is a higher probability that the payload size would be smaller than the `meta` field. And as a consequence.

Reply:
Note that I have extended the field description to add also this idea:

> Imo, it would make sense if this single meta attribute holds all info **an application might want to hand down to the relay layer.**

I explained the field could be used to hand down info to different Waku protocols.

Reply:
If possible, I'd rather move the Message Unique ID (and the potential use cases) to its own PR as this can diverge from the spirit of this PR 🙂